android/src/main/kotlin/io/material/plugins/dynamic_color/DynamicColorPlugin.kt

```diff
@@ -47,6 +56,17 @@ class DynamicColorPlugin : FlutterPlugin, MethodCallHandler {
     channel.setMethodCallHandler(null)
   }
 
+  @RequiresApi(Build.VERSION_CODES.Q)
+  private fun getAccentColor(@NonNull context: Context): Int {
+    val array : TypedArray = context.obtainStyledAttributes(
+      android.R.style.TextAppearance_DeviceDefault,
```

Comment:
This would always tie it to the default theme, no? Have you tried retrieving the resource id from the theme attribute and get the accent color from ContextCompat? Or just not passing in a set theme, do you see differences in colors? (https://developer.android.com/reference/android/content/res/Resources.Theme#obtainStyledAttributes(int[]))

Reply:
What I'm trying to say is that there're different variants of Theme_DeviceDefault, there's also Theme_DeviceDefault_light, Theme_DeviceDefault_daylight, etc. Is the intention here to always retrieve the accent color in dark mode or based on the theme that client have chosen? If you want to retrieve the accent color baased on client's theme instead of only dark mode, do something like this:

```
TypedArray typedArray = contxt.getTheme().obtainStyledAttributes(new int[] {android.R.attr.colorAccent});
int colorAccent = typedArray.getColor(0, /* yourDefValue= */);
typedArray.recycle();
```

If you only want the accent color from dark mode, your solution is fine. Do you see the differences?

Reply:
hmmm that is weird. context.getTheme() should retrieve the current theme, maybe a typo in my code above? I'm not sure...